src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -601,11 +604,29 @@ UCS_CLASS_INIT_FUNC(uct_rdmacm_cm_t, uct_component_h component,
         goto err_destroy_ev_ch;
     }
 
+    if (rdmacm_config->src_addr[0] == '\0') {
+        self->config.src_addr = NULL;
+    } else {
+        self->config.src_addr = ucs_calloc(1, sizeof(struct sockaddr),
+                                           "rdmacm_src_addr");
+        if (self->config.src_addr == NULL) {
+            ucs_error("cannot allocate memory for rdmacm source address");
+            status = UCS_ERR_NO_MEMORY;
+            goto ucs_async_remove_handler;
+        }
+
+        src_af = (strchr(rdmacm_config->src_addr, '.') != NULL) ?
```

Comment:
maybe add an af detection function to sock.c?

Reply:
this is too light weight detection for common code, here I pass address as is to rdmacm without any validation and if user pass wrong addr/format rdmacm would report error